chapters/memory-layout/introduction-to-GDB/reading/README.md

```diff
@@ -0,0 +1,50 @@
+# GNU Debugger (GDB)
+
+## **Starting GDB**
+
+GDB is a powerful tool for debugging programs. It allows you to inspect the state of a program at a certain point in its execution, set breakpoints, and step through the code, among other things. To start GDB, you need to run the following command:
+
+```bash
+gdb [program_name]
+```
+
+## **Running the Program**
+
+To run the program being debugged, there are two available commands:
+
+- `r` or `run` - this command will run the program
+- `start` - unlike `run`, this command will start the program but immediately stop after entering `main`
+
+## **Breakpoints**
+
+The essential element of GDB is the breakpoint. Essentially, setting a breakpoint at a certain instruction causes the program's execution to halt every time it reaches that point. Setting a breakpoint is done with the following command:
+
+```gdb
+break [location]
+```
+
+where *location* can represent the name of a function, the line number of the code, or even a memory address, in which case the address must be preceded by the symbol *. For example: **break \*0xCAFEBABE**
+
+## **Stepping through instructions**
+
+- `si` or `stepi` - executes the current instruction
+- `ni` or `nexti` - similar to `stepi`, but if the current instruction is a function call, the debugger will not enter the function
+- `c` or `continue` - continues program execution until the next breakpoint or until it finishes
+- `finish` - continues program execution until leaving the current function
+
+## **Inspecting Memory**
+
+- `p` or `print` var - displays the value of `var`. Print is a very flexible command, allowing dereferencing of pointers, displaying addresses of variables, and indexing through arrays using *, & and []. The print command can be followed by the /f parameter specifying the display format (x for hex, d for decimal, s for string).
+- `x` or `examine` - Inspects the content at the given address. The usage of this command is as follows:
+
+```py
```

Comment:
Use `text` as the line below isn't actual Python nor does this highlighting make much difference.